src/RefractDataStructure.cc

```diff
@@ -620,54 +628,60 @@ namespace drafter {
             return element;
         }
 
-        void SaveSamples(RefractElements& samples, refract::IElement* element) {
+        template <typename T>
+        struct SaveSamples {
+
+            template <typename U>
+            void operator()(const U& samples, refract::IElement* element) {
 
-            std::for_each(samples.begin(), samples.end(), SetSerializeFlag);
+                if (samples.empty()) {
+                    return;
+                }
 
-            if (!samples.empty()) {
                 refract::ArrayElement* a = new refract::ArrayElement;
-                a->set(samples);
+
+                for (auto sample : samples) {
+                    T* sampleElement = new T;
+                    sampleElement->set(std::get<0>(sample));
+                    SetSerializeFlag(sampleElement);
+                    a->push_back(sampleElement);
+                }
+
                 element->attributes[SerializeKey::Samples] = a;
             }
-        }
 
-        void SaveDefault(RefractElements& defaults, refract::IElement* element) {
+        };
 
-            std::for_each(defaults.begin(), defaults.end(), SetSerializeFlag);
+        template <typename T>
+        struct SaveDefault {
 
-            if (!defaults.empty()) {
-                refract::IElement* e = *defaults.rbegin();
-                defaults.pop_back();
-                // if more default values
-                // use last one, all other we will drop
-                element->attributes[SerializeKey::Default] = e;
+            template <typename U>
+            void operator()(const U& defaults, refract::IElement* element) {
+                if (defaults.empty()) {
+                    return;
+                }
 
-                std::for_each(defaults.begin(), defaults.end(), Deleter<refract::IElement>);
```

Comment:
what are you asking for? 
`Deleter` can be removed from code? - yes, it can

Why it is not required since now?
It is because `ElementData` now used vector of "ValueType" to hold default and samples instead of creating refract elements directly "Extract Functors". Elements (for default samples) are now created later while `TransformElementData` is invoked.